core/src/com/unciv/logic/city/CityInfo.kt

```diff
@@ -645,6 +646,9 @@ class CityInfo {
             .map { it.getOwner()?.civName }.filterNotNull().toSet() 
             .distinct().toList()
     }
+    fun getImprovableTiles(): List<TileInfo> = getTiles().map {it}
+            .filter {it.hasViewableResource(civInfo) && it.improvement == null}.toList()
```

Comment:
Doesn't this also allow bonus resources to be improved? In the original you could only improve strategic & luxury resources iirc

Reply:
Oh i didn't know. Will make change